.circleci/config.yml

```diff
@@ -434,6 +476,111 @@ commands:
             echo $STATUS
             [ $STATUS == 'healthy' ]
 
+  restore_windows_package_cache:
+    parameters:
+      win_cache_version:
+        default: *windows_cache_version
+        type: string
+    steps:
+     - restore_cache:
+        keys:
+          - windows-package-cache-<< parameters.win_cache_version >>-{{ .Branch }}-
+          - windows-package-cache-<< parameters.win_cache_version >>-master-
+          - windows-package-cache-<< parameters.win_cache_version >>-
+
+  save_windows_package_cache:
+    parameters:
+      win_cache_version:
+        type: string
+        default: *windows_cache_version
+      install_path:
+        type: string
+    steps:
+      - save_cache:
+          key: windows-package-cache-<< parameters.win_cache_version >>-{{ .Branch }}-
+          paths:
+            - << parameters.install_path >>
+
+  install_win_msys2:
+    parameters:
+      install_path:
+        default: C:\tools\msys64
+        type: string
+    steps:
+      - restore_windows_package_cache
+      - run:
+          name: Install MSYS2
+          command: .circleci\windows\msys2_prepare.bat
+      - save_windows_package_cache:
+          install_path: << parameters.install_path >>
+
+  restore_windows_erl_cache:
+    parameters:
+      win_cache_version:
+        default: *windows_cache_version
+        type: string
+      erts_version:
+        type: string
+    steps:
+      - restore_cache:
+          keys:
+            - windows-erl<< parameters.erts_version >>-cache-<< parameters.win_cache_version >>-{{ .Branch }}-
+            - windows-erl<< parameters.erts_version >>-cache-<< parameters.win_cache_version >>-master-
+            - windows-erl<< parameters.erts_version >>-cache-<< parameters.win_cache_version >>-
+
+  save_windows_erl_cache:
+    parameters:
+      win_cache_version:
+        default: *windows_cache_version
+        type: string
+      install_path:
+        type: string
+      erts_version:
+        type: string
+    steps:
+      - save_cache:
+          key: windows-erl<< parameters.erts_version >>-cache-<< parameters.win_cache_version >>-{{ .Branch }}-
+          paths:
+            - << parameters.install_path >>
+
+  install_win_otp:
+    parameters:
+      install_path:
+        type: string
+      otp_version:
+        type: string
+      erts_version:
+        type: string
+    steps:
+      - restore_windows_erl_cache:
+          erts_version: << parameters.erts_version >>
+      - run:
+          name: Install Erlang/OTP << parameters.otp_version >>
+          command: scripts\windows\install_erlang << parameters.otp_version >> << parameters.install_path >>
+      - save_windows_erl_cache:
+          erts_version: << parameters.erts_version >>
+          install_path: << parameters.install_path >>
+
+  windows_install_deps:
```

Comment:
OTP and MSYS are split to separate steps, so they can use different caches (this saves some build time) and allows different jobs with different Erlang versions to be cached properly